@here/harp-mapview/lib/copyrights/CopyrightProvider.ts

```diff
@@ -0,0 +1,170 @@
+/*
+ * Copyright (C) 2017-2019 HERE Europe B.V.
+ * Licensed under Apache 2.0, see full license in LICENSE
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { GeoBox } from "@here/harp-geoutils";
+import { getOptionValue, LoggerManager } from "@here/harp-utils";
+import { CopyrightInfo } from "./CopyrightInfo";
+
+declare const require: any;
+
+// tslint:disable-next-line:no-var-requires
+const RBush = require("rbush");
+
+/**
+ * `CopyrightProvider` is an abstract class to retrieve copyrights information for geographic region
+ * specified by bounding box.
+ */
+export abstract class CopyrightProvider {
+    /**
+     * Retrieves copyrights.
+     *
+     * @param geoBox Bounding geo box to get copyrights for.
+     * @param level Zoom level to get copyrights for.
+     * @returns Promise with an array of copyrights for this geo box.
+     */
+    abstract async getCopyrights(geoBox: GeoBox, level: number): Promise<CopyrightInfo[]>;
+}
+
+/** Schema of Map Tile API `copyright` endpoint JSON response. */
```

Comment:
Please link what API is being implemented here, namely: https://developer.here.com/documentation/map-tile/topics/resource-copyright.html
Use markdown, for example like this: [Map Tile API](https://developer.here.com/documentation/map-tile/topics/resource-copyright.html)

Reply:
Done.